### DISCUSSION TITLE
en/simple-implementation-pwa-webdev-mobile-first-design/

### DISCUSSION THREAD

**Discussion by giscus** [idarek/dariusz.wieckiewicz.org#96] [General]
# en/simple-implementation-pwa-webdev-mobile-first-design/

Free world, free thoughts, free words.

https://dariusz.wieckiewicz.org/en/simple-implementation-pwa-webdev-mobile-first-design/

**Comment by iamsukosan** [comment]
Hi, really nice article on PWA, can you also specify for setting-up multiple languages?

Like how can we have manifest files for Spanish and French other than English as main language?

Thanks.

**Reply by idarek** [reply]
Hi, it depends on how you got your website built.
Mine is with [Hugo](https://gohugo.io) and I implement it as follows:

in head.html file (inside `<head>`):
```
  {{ if eq .Lang "en"}}
    <link rel="manifest" href="{{.Site.BaseURL}}manifest.en.webmanifest">
  {{ else }}
    <link rel="manifest" href="{{.Site.BaseURL}}manifest.webmanifest">
  {{ end }}
```

then in `/static` folder, I got `manifest.webmanifest` for the default language and for English I got `manifest.en.webmanifest`.

**Reply by iamsukosan** [reply]
Hi, Dariusz Więckiewicz.

I need help with WordPress hosted website, can you suggest me how can I do it?

**Reply by idarek** [reply]
I am not working with WordPress for some time, but you can have a look into implementing this as a conditional filename that you will be referring to. Assuming that your different language part serves the correct language in `<html lang=` you can have a look at that.

```
<?php $lang=get_bloginfo("language"); ?>
```
and then the output is incorporated into the filename (just a rough idea, not tested)
```
<link rel="manifest" href="/manufest.<?php $lang=get_bloginfo("language"); ?>.webmanifest">
```

Or function

```
function switch_footer_language() {
    if(pll_current_language=='th') {
        <link rel="manifest" href="/manufest.th.webmanifest">
    }
    elseif(pll_current_language=='en') {
        <link rel="manifest" href="/manufest.en.webmanifest">
    }
}
```

**Reply by iamsukosan** [reply]
Hi were you saying something like this?

```
<?php $current_language = get_locale(); ?>
<link rel="manifest" href="/site.<?php ( $current_language == 'en_US' ); ?>.webmanifest">
<link rel="manifest" href="/site.hi-IN.<?php ( $current_language == 'hi_IN' ); ?>.webmanifest">
```

Not working, have any suggestions?

**Comment by iamsukosan** [comment]
Do you think, this code will work?

```
<?php
    $current_language = get_locale();

    if( $current_language == 'en_EN' ){
      echo '/manifest.en.webmanifest';
    }
    
    if( $current_language == 'fr_FR' ){
      echo '/manifest.fr.webmanifest';
    }
    
?>
```

**Comment by idarek** [comment]
Hey @sudarshankotian 
As I mentioned, I am not working with WordPress for some time and don't have a test environment to try.
I give you rough ideas of how this can be implemented. You need to look for a solution that will work for you.

**Reply by iamsukosan** [reply]
No problem, @idarek I got it worked. Thanks for the heads-up.

And your blog was really helpful on PWA.

**Comment by iamsukosan** [comment]
Hi, @idarek how can we manage service worker for multi-language for cache purpose and fallback URL..?

Do you have any idea regarding this?

**Reply by idarek** [reply]
My service worker, as described in the article, is just a dummy file with no extra function hence didn't need to look at that in depth. I will be looking into implementing offline fallback as per [discussion](https://discourse.gohugo.io/t/offline-fallback-hugo-pwa/41188), but in the following weeks when I will have time to do that.

There is only one problem. If somebody adds your website as PWA then even if you change things in the manifest, this will not change for the user unless he removes PWA and creates it again. Unsure if that issue relates to the service workers, but that is what I found out.
